meeteval/io/py.py

```diff
@@ -0,0 +1,254 @@
+        [['abc']]
+        """
+        if self.types is None:
+            raise ValueError('Cannot convert to Python structure because this structure is not invertible.')
+        from meeteval.io.seglst import asseglst
+        t = asseglst(t).map(lambda x: {**defaults, **x})
+        return NestedStructure(_invert_python_structure(t, self.types, self.level_keys + (self.final_key,)),
+                               self.level_keys, self.final_key, self.final_types)
+
+    @property
+    def types(self):
+        if self._seglst is None:
+            self.to_seglst()
+        return self._types
+
+    def to_seglst(self):
+        """
+        Converting Python structures for cpWER, ORC WER and MIMO WER
+        >>> NestedStructure('a b c', level_keys=()).to_seglst()
+        SegLST(segments=[{'words': 'a b c'}])
+
+        Structure levels are interpreted as the keys in keys.
+        >>> NestedStructure('a b c', level_keys=('speaker',)).to_seglst()
+        SegLST(segments=[{'words': 'a b c', 'speaker': 0}])
+        >>> NestedStructure(['a b c', 'd e f'], level_keys=('speaker',)).to_seglst()
+        SegLST(segments=[{'words': 'a b c', 'speaker': 0}, {'words': 'd e f', 'speaker': 1}])
+        >>> NestedStructure({'A': 'a b c', 'B': 'd e f'}, level_keys=('speaker',)).to_seglst()
+        SegLST(segments=[{'words': 'a b c', 'speaker': 'A'}, {'words': 'd e f', 'speaker': 'B'}])
+        >>> NestedStructure({'ex1': {'A': 'a b c', }, 'ex2': {'C': 'd e f'}}, level_keys=('session_id', 'speaker')).to_seglst()
+        SegLST(segments=[{'words': 'a b c', 'speaker': 'A', 'session_id': 'ex1'}, {'words': 'd e f', 'speaker': 'C', 'session_id': 'ex2'}])
+
+        All keys in `keys` must be present, otherwise an exception is raised
+        >>> NestedStructure('a b c', level_keys=('speaker', 'channel')).to_seglst()
+        Traceback (most recent call last):
+          ...
+        ValueError: Key mismatch
+
+        Empty structures
+
+        Empty structures result in empty `SegLST` objects
+        >>> NestedStructure([], level_keys=('speaker',)).to_seglst()
+        SegLST(segments=[])
+        >>> NestedStructure({}, level_keys=('speaker',)).to_seglst()
+        SegLST(segments=[])
+        >>> NestedStructure([{}], level_keys=('speaker',)).to_seglst()
+        SegLST(segments=[])
+
+        Empty nested structures are allowed but not represented in the `SegLST` format (TODO)
+        >>> NestedStructure([['ab'], []], level_keys=('speaker', 'segment_index')).to_seglst()
+        SegLST(segments=[{'words': 'ab', 'segment_index': 0, 'speaker': 0}])
+
+        The last key is filled with a dummy key
+        >>> NestedStructure(['a b c', 'd e f'], level_keys=('speaker', 'channel')).to_seglst()
+        SegLST(segments=[{'words': 'a b c', 'channel': 0, 'speaker': 0}, {'words': 'd e f', 'channel': 0, 'speaker': 1}])
+
+        Nested structures beyond the specified groups are by default not allowed. With `ensure_word_is_string=False`,
```

Comment:
An example would be a generalized `apply_assignment` where you can input, for example `[SomeObject(), SomeObject()]`. But currently it's unused and should not be used directly by the user (I think) because it can lead to confusing behavior

Reply:
So it is already used in some code?

At the moment, the input may allow too much.
The `final_types` argument may be difficult to use, because it works only on a finite set of the input structure.

Reply:
It is currently unused. I had a draft of a generalized apply_assignment function, but that turned out to be too complicated to push so I removed it. I can remove the `final_types` arg as long as it is not needed

Reply:
We could rename that argument to `_final_types` (i.e. make it private) and make a statement, that it is WIP for a generalized apply_assignment function.
It may turn out later, that we want to implement it differently to what `final_types` does.

In general:
When we allow less, we have later more freedom to change something to add new features.

Reply:
Okay, I made it private